granulate_utils/containers/docker.py

```diff
@@ -35,8 +36,23 @@ def get_runtimes(self) -> List[str]:
         return ["docker"]
 
     @staticmethod
-    def _create_container(container: docker.models.containers.Container) -> Container:
+    def _parse_docker_timestamp(time_str: str) -> Optional[datetime]:
+        """
+        Parses timestamps provided by DockerAPI to datetime.
+        Docker works with ISO format timestamps (in UTC) with fractional milliseconds that python standard library
+        doesn't parse, and also ends with "Z" timezone indicator for UTC.
+        """
+        assert time_str.endswith('Z')  # assert UTC
+        if time_str.startswith('0001'):
+            return None
+        return datetime.fromisoformat(time_str.split('.')[0])
```

Comment:
Have you tried with `dateutil`?

Reply:
dateutil will succeed here indeed

Reply:
Will probably work, but I didn't want to introduce another requirement

Reply:
Yes but its redundant

Reply:
> Yes but its redundant

Why is it redundant?

Reply:
Because the user knows its UTC because this is what I wrote in the comment next to the class var definition :)

Reply:
any operation on a datetime object that does not contain timezone info assumes its in local time. it does not matter what the user knows, the code in the datetime module doesnt